File: src/main/java/com/leijendary/spring/boot/template/json/deserializer/StringTrimDeserializer.java
package com.leijendary.spring.boot.template.json.deserializer;

import static java.util.Optional.ofNullable;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StringDeserializer;

import org.springframework.boot.jackson.JsonComponent;

@JsonComponent
public class StringTrimDeserializer extends StringDeserializer {

    @Override
    public String deserialize(final JsonParser jsonParser, final DeserializationContext deserializationContext)
            throws IOException {
        // Get the result from the original StringDeserializer
        final var deserialized = super.deserialize(jsonParser, deserializationContext);

        // ... and just trim it
        return ofNullable(deserialized)
                .map(String::trim)
                .orElse(deserialized);
    }
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/json/StringTrimDeserializer.kt
package com.leijendary.spring.boot.template.core.json

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StringDeserializer
import org.springframework.boot.jackson.JsonComponent
import java.io.IOException

@JsonComponent
class StringTrimDeserializer : StringDeserializer() {
    @Throws(IOException::class)
    override fun deserialize(jsonParser: JsonParser, deserializationContext: DeserializationContext): String? {
        // Get the result from the original StringDeserializer
        val deserialized = super.deserialize(jsonParser, deserializationContext)

        // ... and just trim it
        return deserialized?.let { obj: String -> obj.trim { it <= ' ' } }
    }
}